server/scsynth/SC_GraphDef.cpp

```diff
@@ -604,10 +604,10 @@ SCErr GraphDef_DeleteMsg(World *inWorld, GraphDef *inDef)
 	packet.addtag('s');
 	packet.adds((char*)inDef->mNodeDef.mName);
 
-	ReplyAddress *users = inWorld->hw->mUsers;
-	int numUsers = inWorld->hw->mNumUsers;
-	for (int i=0; i<numUsers; ++i) {
-		SCErr err = SendReplyCmd_d_removed(inWorld, packet.size(), packet.data(), users+i);
+	Clients::iterator it;
+	for (it = inWorld->hw->mUsers->begin(); it != inWorld->hw->mUsers->end(); ++it){
```

Comment:
all the c++ code here is merged in from @muellmusik's branch : https://github.com/supercollider/supercollider/tree/topic/improved-server-notify 
@muellmusik, can you answer these please? 
my c++ skills are nowhere near good enough to improve this code.